platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/module/telemetry/TelemetryImpl.java

```diff
@@ -109,6 +112,26 @@ public boolean setSessionIdRotationInterval(int interval) {
     return telemetry.updateSessionIdRotationInterval(new SessionInterval(interval));
   }
 
+  @Override
+  public void onCreateOfflineRegion(OfflineRegionDefinition offlineDefinition) {
+    MapEventFactory mapEventFactory = new MapEventFactory();
+
+    Event mapOfflineEvent;
+    if (offlineDefinition instanceof OfflineTilePyramidRegionDefinition) {
+      OfflineTilePyramidRegionDefinition tileDefinition =
+        (OfflineTilePyramidRegionDefinition)offlineDefinition;
+      mapOfflineEvent = mapEventFactory.buildMapOfflineEvent(
+        tileDefinition.getMinZoom(), tileDefinition.getMaxZoom(), "bounds",
+        new String[]{tileDefinition.getStyleURL()});
+    } else {
+      OfflineGeometryRegionDefinition geometryDefinition =
+        (OfflineGeometryRegionDefinition) offlineDefinition;
+      mapOfflineEvent = mapEventFactory.buildMapOfflineEvent(
+        geometryDefinition.getMinZoom(), geometryDefinition.getMaxZoom(), geometryDefinition.getGeometry().type(),
+        new String[]{geometryDefinition.getStyleURL()});
```

Comment:
Just as a heads up we're collecting the style URL here and not a list of sources as described in the current schema noted in https://github.com/mapbox/event-schema/pull/64.

Per chat with @1ec5 and @chloekraw it would be more useful to log the sources belonging to the style rather than the style url itself, so I'm working with the core team to see if there is a way we can expose this easily. You might need to amend this PR or create a new one if we are able to accomplish this.

Reply:
@captainbarbosa  I see. Do you think I should pass `null` right now or just leave it as is ?